tests/integration/cli/restart_test.py

```diff
@@ -0,0 +1,71 @@
+import datetime
+import os
+import time
+
+import dateutil.parser
+from nose.tools import nottest
+
+from dusty.compiler.spec_assembler import get_all_repos
+from dusty.source import Repo
+from ...testcases import DustyIntegrationTestCase
+from ...fixtures import specs_fixture_with_depends
+
+class TestRestartCLI(DustyIntegrationTestCase):
+    def setUp(self):
+        super(TestRestartCLI, self).setUp()
+        specs_fixture_with_depends()
+        for repo in get_all_repos(include_specs_repo=False):
+            self._set_up_fake_local_repo(path=repo.remote_path)
+        self.run_command('bundles activate bundle-a bundle-b')
+        self.run_command('up')
+        time.sleep(1)
+        self.up_complete_time = datetime.datetime.utcnow()
+
+    def tearDown(self):
+        try:
+            self.run_command('stop')
+        except Exception:
+            pass
+        super(TestRestartCLI, self).tearDown()
+
+    @nottest
+    def container_has_restarted(self, app_name):
+        inspected = self.inspect_container(app_name)
+        start_time = dateutil.parser.parse(inspected['State']['StartedAt']).replace(tzinfo=None)
+        return start_time > self.up_complete_time
+
+    def test_restart_one(self):
+        self.run_command('restart appa')
+        self.assertTrue(self.container_has_restarted('appa'))
+        self.assertTrue(not self.container_has_restarted('appb'))
+
+    def test_restart_all(self):
+        self.run_command('restart')
+        self.assertTrue(self.container_has_restarted('appa'))
+        self.assertTrue(self.container_has_restarted('appb'))
+
+    def test_restart_by_app_repo(self):
+        self.run_command('restart --repos app-a')
+        self.assertTrue(self.container_has_restarted('appa'))
+        self.assertTrue(not self.container_has_restarted('appb'))
+
+    def test_restart_by_lib_repo(self):
+        self.run_command('restart --repos lib-a')
+        self.assertTrue(self.container_has_restarted('appa'))
+        self.assertTrue(self.container_has_restarted('appb'))
```

Comment:
Might be worth it to add an app that doesn't depend on this lib and confirm that it hasn't restarted